pkgs/tools/games/minecraft/optifine/update.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python3 -p python3 python3.pkgs.requests python3.pkgs.lxml
+
+from lxml import html
+import json
+import os.path
+import re
+import requests
+import subprocess
+
+def nix_prefetch_sha256(name):
+    return subprocess.run(['nix-prefetch-url', '--type', 'sha256', 'https://optifine.net/download?f=' + name], capture_output=True, text=True).stdout.strip()
+
+# fetch download page
+print('Looking up `https://optifine.net/downloads` for versions.')
+sess = requests.session()
+page = sess.get('https://optifine.net/downloads')
+tree = html.fromstring(page.content)
+
+# parse and extract main jar file names
+href = tree.xpath('//tr[@class="downloadLine downloadLineMain"]/td[@class="colMirror"]/a/@href')
+prog = re.compile('(OptiFine_)([0-9.]*)(.*)\.jar')
```

Comment:
Is `prog` a standardized name? I'm not sure what it stands for here.

Reply:
Ha! No, it's just that I don't really know what to call a "[regular expression object](https://docs.python.org/3/library/re.html#re-objects)", so I just named the variable `prog` as in "program". I have changed it to `expr` so hopefully it makes a little more sense.